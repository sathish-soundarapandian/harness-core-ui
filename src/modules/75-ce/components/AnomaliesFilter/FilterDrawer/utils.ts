/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { MultiSelectOption } from '@harness/uicore'
import type { AnomalyFilterProperties } from 'services/ce'
import type { AnomaliesFilterFormType } from './FilterDrawer'

const getMultiSelectOptions = (values: string[]): MultiSelectOption[] =>
  values.map(item => ({ label: item, value: item }))

const getValueFromOption = (values: MultiSelectOption[]): string[] => values.map(item => item.value.toString())

export const getAnomalyFormValuesFromFilterProperties = (
  filterProperties: AnomalyFilterProperties
): AnomaliesFilterFormType => {
  const formValues: AnomaliesFilterFormType = {}

  const {
    awsAccounts,
    awsServices,
    awsUsageTypes,
    gcpProducts,
    gcpProjects,
    gcpSKUDescriptions,
    k8sClusterNames,
    k8sNamespaces,
    k8sWorkloadNames,
    azureResourceGroups,
    azureMeterCategories,
    azureSubscriptionGuids,
    minActualAmount,
    minAnomalousSpend
  } = filterProperties

  if (awsAccounts) {
    formValues.awsAccounts = getMultiSelectOptions(awsAccounts)
  }
  if (awsServices) {
    formValues.awsServices = getMultiSelectOptions(awsServices)
  }
  if (awsUsageTypes) {
    formValues.awsUsageTypes = getMultiSelectOptions(awsUsageTypes)
  }
  if (gcpProducts) {
    formValues.gcpProducts = getMultiSelectOptions(gcpProducts)
  }
  if (gcpProjects) {
    formValues.gcpProjects = getMultiSelectOptions(gcpProjects)
  }
  if (gcpSKUDescriptions) {
    formValues.gcpSKUDescriptions = getMultiSelectOptions(gcpSKUDescriptions)
  }
  if (k8sClusterNames) {
    formValues.k8sClusterNames = getMultiSelectOptions(k8sClusterNames)
  }
  if (k8sNamespaces) {
    formValues.k8sNamespaces = getMultiSelectOptions(k8sNamespaces)
  }
  if (k8sWorkloadNames) {
    formValues.k8sWorkloadNames = getMultiSelectOptions(k8sWorkloadNames)
  }
  if (azureResourceGroups) {
    formValues.azureResourceGroups = getMultiSelectOptions(azureResourceGroups)
  }
  if (azureMeterCategories) {
    formValues.azureMeterCategories = getMultiSelectOptions(azureMeterCategories)
  }
  if (azureSubscriptionGuids) {
    formValues.azureSubscriptionGuids = getMultiSelectOptions(azureSubscriptionGuids)
  }

  if (minActualAmount) {
    formValues.minActualAmount = +minActualAmount
  }

  if (minAnomalousSpend) {
    formValues.minAnomalousSpend = +minAnomalousSpend
  }

  return formValues
}

export const getAnomalyFilterPropertiesFromForm = (formData: AnomaliesFilterFormType): AnomalyFilterProperties => {
  const filterProperties: AnomalyFilterProperties = { filterType: 'Anomaly' }

  Object.entries(formData).forEach(([key, value]) => {
    if (key === 'minActualAmount' || key === 'minAnomalousSpend') {
      filterProperties[key] = +value
    } else {
      ;(filterProperties as any)[key] = getValueFromOption(value)
    }
  })

  return filterProperties
}
