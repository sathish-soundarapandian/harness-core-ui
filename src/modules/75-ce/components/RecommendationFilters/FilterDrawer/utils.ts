/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { CCMRecommendationFilterProperties, K8sRecommendationFilterPropertiesDTO } from 'services/ce'
import type { RecommendationFilterFormType } from '@ce/components/RecommendationFilters/FilterDrawer/FilterDrawer'

export const getRecommendationFormValuesFromFilterProperties = (
  filterProperties: CCMRecommendationFilterProperties
): RecommendationFilterFormType => {
  const formData: RecommendationFilterFormType = {}

  const { k8sRecommendationFilterPropertiesDTO = {}, minCost, minSaving, perspectiveFilters } = filterProperties
  const { clusterNames, names, namespaces, resourceTypes } = k8sRecommendationFilterPropertiesDTO

  if (clusterNames) {
    formData.clusterNames = clusterNames.map(clusterName => ({ label: clusterName, value: clusterName }))
  }

  if (names) {
    formData.names = names.map(name => ({ label: name, value: name }))
  }

  if (namespaces) {
    formData.namespaces = namespaces.map(namespace => ({ label: namespace, value: namespace }))
  }

  if (resourceTypes) {
    formData.resourceTypes = resourceTypes.map(resourceType => ({ label: resourceType, value: resourceType }))
  }

  if (minCost) {
    formData.minCost = minCost
  }

  if (minSaving) {
    formData.minSaving = minSaving
  }

  if (perspectiveFilters) {
    formData.perspectiveFilters = perspectiveFilters
  }

  return formData
}

export const getRecommendationFilterPropertiesFromForm = (
  formData: RecommendationFilterFormType
): CCMRecommendationFilterProperties => {
  const filterProperties: CCMRecommendationFilterProperties = { filterType: 'CCMRecommendation' }
  const { clusterNames, minCost, minSaving, names, namespaces, resourceTypes, perspectiveFilters } = formData

  const k8sRecommendationFilterPropertiesDTO = {} as K8sRecommendationFilterPropertiesDTO
  if (clusterNames) {
    k8sRecommendationFilterPropertiesDTO.clusterNames = clusterNames.map(item => item.value) as string[]
  }
  if (namespaces) {
    k8sRecommendationFilterPropertiesDTO.namespaces = namespaces.map(item => item.value) as string[]
  }
  if (names) {
    k8sRecommendationFilterPropertiesDTO.names = names.map(item => item.value) as string[]
  }
  if (resourceTypes) {
    k8sRecommendationFilterPropertiesDTO.resourceTypes = resourceTypes.map(
      item => item.value
    ) as K8sRecommendationFilterPropertiesDTO['resourceTypes']
  }
  if (minCost) {
    filterProperties.minCost = +minCost
  }

  if (minSaving) {
    filterProperties.minSaving = +minSaving
  }

  if (k8sRecommendationFilterPropertiesDTO) {
    filterProperties.k8sRecommendationFilterPropertiesDTO = k8sRecommendationFilterPropertiesDTO
  }

  if (perspectiveFilters) {
    filterProperties.perspectiveFilters = perspectiveFilters
  }

  return filterProperties
}
