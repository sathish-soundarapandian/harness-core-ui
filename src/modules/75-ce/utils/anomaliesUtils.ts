/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { CloudProvider } from '@ce/types'
import type { QlceViewFilterInput } from 'services/ce/services'
import { DEFAULT_GROUP_BY } from './perspectiveUtils'

const gcpEntities = ['gcpProduct', 'gcpProjectId', 'gcpSKUDescription']
const awsEntities = ['awsUsageAccountId', 'awsServiceCode', 'awsInstancetype', 'awsUsageType']
const azureEntities = ['azureSubscriptionGuid', 'azureMeterCategory', 'azureResourceGroup']
const clusterEntities = ['clusterName', 'namespace', 'workloadName', 'workloadType']

const cloudProviderToEntityMapping = {
  [CloudProvider.GCP]: gcpEntities,
  [CloudProvider.AWS]: awsEntities,
  [CloudProvider.CLUSTER]: clusterEntities,
  [CloudProvider.AZURE]: azureEntities
}

export const anomalyFilterValueColumns = [
  'clustername',
  'namespace',
  'workloadname',
  'gcpproject',
  'gcpproduct',
  'gcpskudescription',
  'awsaccount',
  'awsservice',
  'awsusagetype',
  'azuresubscriptionguid',
  'azureresourcegroup',
  'azuremetercategory'
]

export const filterKeyToKeyMapping: Record<string, string> = {
  gcpProducts: 'gcpproduct',
  gcpProjects: 'gcpproject',
  gcpSKUDescriptions: 'gcpskudescription',
  k8sClusterNames: 'clustername',
  k8sNamespaces: 'namespace',
  k8sWorkloadNames: 'workloadname',
  awsAccounts: 'awsaccount',
  awsServices: 'awsservice',
  awsUsageTypes: 'awsusagetype',
  azureSubscriptionGuids: 'azuresubscriptionguid',
  azureResourceGroups: 'azureresourcegroup',
  azureMeterCategories: 'azuremetercategory'
}

export const filterKeyToLabelMapping: Record<string, string> = {
  gcpProducts: 'GCP Product',
  gcpProjects: 'GCP Project',
  gcpSKUDescriptions: 'GCP SKU Description',
  k8sClusterNames: 'Cluster Name',
  k8sNamespaces: 'Namespace',
  k8sWorkloadNames: 'Workload',
  awsAccounts: 'AWS Account',
  awsServices: 'AWS Service',
  awsUsageTypes: 'AWS Usage Type',
  azureSubscriptionGuids: 'AZURE Subscription',
  azureMeterCategories: 'AZURE Meter Categories',
  azureResourceGroups: 'AZURE Resource'
}

export const k8sFilterKeys = ['k8sClusterNames', 'k8sNamespaces', 'k8sWorkloadNames']
export const gcpFilterKeys = ['gcpProjects', 'gcpProducts', 'gcpSKUDescriptions']
export const awsFilterKeys = ['awsAccounts', 'awsServices', 'awsUsageTypes']
export const azureFilterKeys = ['azureSubscriptionGuids', 'azureMeterCategories', 'azureResourceGroups']

const fieldToFieldNameMapping: Record<string, string> = {
  gcpProduct: 'Product',
  gcpProjectId: 'Project',
  gcpSKUDescription: 'SKUs',
  clusterName: 'Cluster Name',
  namespace: 'Namespace',
  workloadName: 'Workload',
  awsUsageAccountId: 'Account',
  awsServiceCode: 'Service',
  awsInstancetype: 'Instance Type',
  awsUsageType: 'Usage Type',
  workloadType: 'Workload Type',
  gcpSkuDescription: 'SKUs',
  azureSubscriptionGuid: 'Subscription ID',
  azureMeterCategory: 'Meter category',
  azureResourceGroup: 'Resource group name'
}

export function generateFilters(
  entityMap: Record<string, string>,
  cloudProvider: CloudProvider
): QlceViewFilterInput[] {
  const filters: any = []
  const relatedEntities = cloudProviderToEntityMapping[cloudProvider]
  relatedEntities?.length &&
    relatedEntities.forEach(entity => {
      if (entityMap[entity]) {
        filters.push({
          field: {
            fieldId: entity,
            fieldName: fieldToFieldNameMapping[entity] || '',
            identifier: cloudProvider,
            identifierName: cloudProvider
          },
          operator: 'IN',
          type: 'VIEW_ID_CONDITION',
          values: [entityMap[entity]]
        })
      }
    })

  return filters
}

export function generateGroupBy(field: string, cloudProvider: CloudProvider): any {
  return field
    ? {
        fieldId: field,
        fieldName: fieldToFieldNameMapping[field] || '',
        identifier: cloudProvider,
        identifierName: cloudProvider
      }
    : DEFAULT_GROUP_BY
}

export function getCloudProviderFromFields(entityMap: Record<string, string | null>): CloudProvider {
  if (entityMap.gcpProjectId) {
    return CloudProvider.GCP
  }
  if (entityMap.awsUsageAccountId) {
    return CloudProvider.AWS
  }
  if (entityMap.azureSubscriptionGuid) {
    return CloudProvider.AZURE
  }
  return CloudProvider.CLUSTER
}

export function getFiltersFromEnityMap(
  entityMap: Array<Record<string, string>>,
  cloudProvider: CloudProvider
): QlceViewFilterInput[] {
  const filters: any = []
  const relatedEntities = cloudProviderToEntityMapping[cloudProvider]

  relatedEntities.length &&
    relatedEntities.forEach(entity => {
      if (entityMap.length && entityMap[0][entity]) {
        const values: string[] = []
        entityMap.forEach(obj => {
          obj[entity] && values.push(obj[entity])
        })

        values.length &&
          filters.push({
            field: {
              fieldId: entity,
              fieldName: fieldToFieldNameMapping[entity] || '',
              identifier: cloudProvider,
              identifierName: cloudProvider
            },
            operator: 'IN',
            type: 'VIEW_ID_CONDITION',
            values: values
          })
      }
    })

  return filters
}
