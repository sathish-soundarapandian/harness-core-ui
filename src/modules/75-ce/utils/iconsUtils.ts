/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/icons'
import { CloudProvider } from '@ce/types'

export const getResourceIcon = (cloudProvider: string): IconName => {
  switch (cloudProvider) {
    case CloudProvider.CLUSTER:
      return 'service-kubernetes'

    case CloudProvider.AWS:
      return 'service-aws'

    case CloudProvider.AZURE:
      return 'service-azure'

    case CloudProvider.GCP:
      return 'gcp'

    default:
      return 'harness'
  }
}

export const getServiceIcons = (serviceProvider: string): IconName => {
  switch (serviceProvider) {
    case 'RDS':
      return 'aws-rds'

    case 'GCP INSTANCE':
    case 'GCP DISK':
      return 'gcp-engine'

    case 'AZURE VM':
      return 'azure-vm'

    case 'AWS EC2':
    case 'AWS EBS':
      return 'aws-ectwo-service'

    default:
      return 'default-dashboard'
  }
}
