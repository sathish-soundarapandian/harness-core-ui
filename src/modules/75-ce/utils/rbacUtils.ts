/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export const getPermissionErrorMsg = (errorMessage: string): string => {
  const colonIndex: number = errorMessage.indexOf(':')
  if (colonIndex > -1) {
    return errorMessage.slice(colonIndex + 1).trim()
  }
  return errorMessage
}
