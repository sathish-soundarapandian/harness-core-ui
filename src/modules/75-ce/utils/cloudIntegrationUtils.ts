/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { Color } from '@harness/design-system'
import type { IconName, IconProps } from '@harness/icons'
import qs from 'qs'
import type { CellProps, Column, Renderer } from 'react-table'

import type {
  CcmK8sConnectorResponse,
  ConnectorResponse,
  PageCcmK8sConnectorResponse,
  ConnectorValidationResult,
  ConnectorInfoDTO
} from 'services/cd-ng'
import type { CcmK8sMetaInfo, CcmK8sMetaInfoResponseDTO } from 'services/ce'
import type { CcmMetaData } from 'services/ce/services'
import routes from '@common/RouteDefinitions'
import { CloudProvider } from '@ce/types'
import { generateFilters } from './anomaliesUtils'

export const getConnectorStatusIcon: (status: ConnectorValidationResult['status']) => {
  icon: IconName
  iconProps: Partial<IconProps>
} = status =>
  status === 'SUCCESS'
    ? { icon: 'full-circle', iconProps: { size: 8, color: Color.GREEN_500 } }
    : { icon: 'warning-sign', iconProps: { size: 12, color: Color.RED_500 } }

/**
 * Kubernetes Clusters Tab
 */

export type CustomK8sConnectorResponse = CcmK8sConnectorResponse & {
  metadata?: CcmK8sMetaInfo
}

export type CustomK8sPageConnectorResponse = PageCcmK8sConnectorResponse & {
  content?: {
    metadata?: CcmK8sMetaInfo
  }
}

export type CustomK8sColumn = Column<CustomK8sConnectorResponse>[]
export type CustomK8sCell = Renderer<CellProps<CustomK8sConnectorResponse>>

export const mapCCMK8sMetadataToConnector = (
  connectors?: PageCcmK8sConnectorResponse,
  ccmK8sMetadata?: CcmK8sMetaInfoResponseDTO
): CustomK8sPageConnectorResponse => {
  return {
    ...connectors,
    content: connectors?.content?.map(item => ({
      ...item,
      metadata: ccmK8sMetadata?.ccmK8sMeta?.find(
        meta => meta.ccmk8sConnectorId === item.ccmk8sConnector?.[0]?.connector?.identifier
      )
    }))
  }
}

export const getSuccesfullCCMConnectorIds = (connectors?: PageCcmK8sConnectorResponse): string[] =>
  connectors?.content
    ?.filter(item => item.ccmk8sConnector?.[0]?.status?.status === 'SUCCESS')
    .map(item => item.ccmk8sConnector?.[0]?.connector?.identifier)
    .filter(item => item) as string[]

export const getReportingTooltipIcon = (
  isConnectorStatusSuccess: boolean,
  isLastEventRecieved: boolean
): { name: IconName; color: Color } => {
  if (!isConnectorStatusSuccess) {
    return { name: 'warning-sign', color: Color.RED_700 }
  }
  if (!isLastEventRecieved) {
    return { name: 'heart-broken', color: Color.RED_700 }
  }

  return { name: 'heart', color: Color.GREEN_700 }
}

export const lastEventRecieved = (metadata?: CcmK8sMetaInfo): boolean =>
  !!metadata?.visibility?.[0]?.includes('last event received at')

/**
 * Cloud Accounts Tab
 */

export type CustomCloudColumn = Column<ConnectorResponse>[]
export type CustomCloudCell = Renderer<CellProps<ConnectorResponse>>

const defaultPerspectiveIdMap: Record<string, keyof CcmMetaData> = {
  CEAzure: 'defaultAzurePerspectiveId',
  CEAws: 'defaultAwsPerspectiveId',
  GcpCloudCost: 'defaultGcpPerspectiveId'
}

const connectorTypeToCloudProvider: Record<string, CloudProvider> = {
  CEAzure: CloudProvider.AZURE,
  CEAws: CloudProvider.AWS,
  GcpCloudCost: CloudProvider.GCP
}

export const getCloudViewCostsLink = ({
  accountId,
  connector,
  ccmMetaData
}: {
  accountId: string
  connector?: ConnectorInfoDTO
  ccmMetaData?: CcmMetaData | null
}): string | undefined => {
  const defaultPerspectiveId = ccmMetaData?.[defaultPerspectiveIdMap[connector?.type || '']] as string

  if (!defaultPerspectiveId) {
    return
  }

  let filters = {}

  switch (connector?.type) {
    case 'CEAzure': {
      filters = { azureSubscriptionGuid: connector?.spec?.subscriptionId }
      break
    }
    case 'CEAws': {
      filters = { awsUsageAccountId: connector?.spec?.awsAccountId }
      break
    }
    case 'GcpCloudCost': {
      filters = { gcpProjectId: connector?.spec?.projectId }
    }
  }

  return (
    routes.toPerspectiveDetails({
      accountId: accountId,
      perspectiveId: defaultPerspectiveId,
      perspectiveName: defaultPerspectiveId
    }) +
    `?${qs.stringify({
      filters: JSON.stringify(generateFilters(filters, connectorTypeToCloudProvider[connector?.type || '']))
    })}`
  )
}
