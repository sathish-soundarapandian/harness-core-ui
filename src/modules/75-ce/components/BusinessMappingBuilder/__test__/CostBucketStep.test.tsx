/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import {} from '@harness/uicore'
import { Color } from '@harness/design-system'
import { TestWrapper } from '@common/utils/testUtils'
import { CostBucketWidgetType } from '@ce/types'
import CostBucketStep from '../CostBucketStep/CostBucketStep'

const value = [
  {
    name: 'Rule1',
    rules: [
      {
        viewConditions: [
          {
            values: ['d', 'e', 'f'],
            viewOperator: 'IN',
            viewField: {
              fieldId: 'product',
              fieldName: 'Product',
              identifier: 'COMMON',
              identifierName: 'Common'
            }
          }
        ]
      }
    ],
    strategy: 'Fixed'
  }
]

describe('test cases for Cost Bucket Step', () => {
  test('should be able to render', () => {
    const props: any = {
      formikProps: {
        values: {
          sharedCostBucketKey: 1
        },
        setFieldValue: jest.fn()
      },
      value: value,
      namespace: 'sharedCostBucket',
      isSharedCost: true,
      widgetType: CostBucketWidgetType.SharedCostBucket,
      stepProps: {
        color: Color.PRIMARY_7,
        background: Color.PRIMARY_1,
        total: 3,
        current: 1,
        defaultOpen: true
      }
    }

    const { container } = render(
      <TestWrapper>
        <CostBucketStep {...props} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
