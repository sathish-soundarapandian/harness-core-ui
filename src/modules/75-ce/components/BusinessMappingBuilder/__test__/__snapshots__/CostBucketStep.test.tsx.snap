// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for Cost Bucket Step should be able to render 1`] = `
<div>
  <div
    class="bp3-card bp3-elevation-0 Card--card container"
  >
    <div
      class="StyledProps--font StyledProps--main"
    >
      <div
        class="Layout--horizontal StyledProps--main headerContainer"
      >
        <p
          class="StyledProps--font StyledProps--main pillRadius StyledProps--color StyledProps--color-primary7 StyledProps--background-primary1 StyledProps--padding-top-xsmall StyledProps--padding-bottom-xsmall StyledProps--padding-left-small StyledProps--padding-right-small StyledProps--font-variation-tiny"
        >
          ce.businessMapping.stepText
        </p>
        <span
          style="flex-grow: 1;"
        />
        <button
          class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-chevron-up StyledProps--main"
            icon="chevron-up"
          >
            <svg
              data-icon="chevron-up"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <desc>
                chevron-up
              </desc>
              <path
                d="M16.71 12.29l-6-6C10.53 6.11 10.28 6 10 6s-.53.11-.71.29l-6 6a1.003 1.003 0 001.42 1.42L10 8.41l5.29 5.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="Layout--horizontal StyledProps--main headerContainer"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--padding-top-small StyledProps--font-variation-card-title"
        >
          ce.businessMapping.sharedCostBucket.title
        </p>
        <span
          style="flex-grow: 1;"
        />
        <button
          aria-label="ce.businessMapping.sharedCostBucket.newButtonText"
          class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--without-shadow Button--withLeftIcon Button--size Button--small"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-plus StyledProps--main StyledProps--padding-right-xsmall"
            icon="plus"
          >
            <svg
              data-icon="plus"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                plus
              </desc>
              <path
                d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            ce.businessMapping.sharedCostBucket.newButtonText
          </span>
        </button>
      </div>
      <div
        class="bp3-collapse"
        style="height: auto; overflow-y: visible; transition: none;"
      >
        <div
          aria-hidden="false"
          class="bp3-collapse-body"
          style="transform: translateY(0); transition: none;"
        >
          <div
            class="StyledProps--font StyledProps--main StyledProps--margin-top-medium StyledProps--border StyledProps--border-top"
          >
            <div>
              <div
                class="StyledProps--font StyledProps--main drag"
                data-rbd-draggable-context-id="0"
                data-rbd-draggable-id="sharedCostBucket-0-Rule1"
              >
                <div
                  class="bp3-card bp3-elevation-0 Card--card mainContainer"
                >
                  <div
                    class="Layout--horizontal StyledProps--main costBucketNameContainet StyledProps--background-grey100"
                  >
                    <div
                      aria-describedby="rbd-hidden-text-0-hidden-text-0"
                      class="StyledProps--font StyledProps--main"
                      data-rbd-drag-handle-context-id="0"
                      data-rbd-drag-handle-draggable-id="sharedCostBucket-0-Rule1"
                      draggable="false"
                      role="button"
                      tabindex="0"
                    >
                      <span
                        class="bp3-icon bp3-icon-drag-handle-vertical StyledProps--main StyledProps--margin-top-xsmall StyledProps--color StyledProps--color-grey300"
                        icon="drag-handle-vertical"
                      >
                        <svg
                          data-icon="drag-handle-vertical"
                          height="24"
                          viewBox="0 0 20 20"
                          width="24"
                        >
                          <desc>
                            drag-handle-vertical
                          </desc>
                          <path
                            d="M7.5 6C6.67 6 6 6.67 6 7.5S6.67 9 7.5 9 9 8.33 9 7.5 8.33 6 7.5 6zm0 5c-.83 0-1.5.67-1.5 1.5S6.67 14 7.5 14 9 13.33 9 12.5 8.33 11 7.5 11zm0 5c-.83 0-1.5.67-1.5 1.5S6.67 19 7.5 19 9 18.33 9 17.5 8.33 16 7.5 16zm5-12c.83 0 1.5-.67 1.5-1.5S13.33 1 12.5 1 11 1.67 11 2.5 11.67 4 12.5 4zm-5-3C6.67 1 6 1.67 6 2.5S6.67 4 7.5 4 9 3.33 9 2.5 8.33 1 7.5 1zm5 10c-.83 0-1.5.67-1.5 1.5s.67 1.5 1.5 1.5 1.5-.67 1.5-1.5-.67-1.5-1.5-1.5zm0 5c-.83 0-1.5.67-1.5 1.5s.67 1.5 1.5 1.5 1.5-.67 1.5-1.5-.67-1.5-1.5-1.5zm0-10c-.83 0-1.5.67-1.5 1.5S11.67 9 12.5 9 14 8.33 14 7.5 13.33 6 12.5 6z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </div>
                    <div
                      class="bp3-form-group nameInput"
                    >
                      <div
                        class="bp3-form-content"
                      >
                        <div
                          class="bp3-input-group"
                        >
                          <input
                            autocomplete="off"
                            class="bp3-input"
                            name="sharedCostBucket[0].name"
                            placeholder="ce.businessMapping.sharedCostBucket.inputName"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                    <span
                      style="flex-grow: 1;"
                    />
                    <button
                      class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main buttonNoPadding StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
                      type="button"
                    >
                      <span
                        class="bp3-icon StyledProps--main"
                        data-icon="Edit"
                      >
                        <svg
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                          <path
                            clip-rule="evenodd"
                            d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                            fill="currentColor"
                            fill-rule="evenodd"
                            opacity="0.3"
                          />
                        </svg>
                      </span>
                    </button>
                    <button
                      class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main buttonNoPadding StyledProps--intent-primary Button--with-current-color Button--without-shadow Button--withLeftIcon"
                      type="button"
                    >
                      <span
                        class="bp3-icon bp3-icon-chevron-down StyledProps--main"
                        icon="chevron-down"
                      >
                        <svg
                          data-icon="chevron-down"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <desc>
                            chevron-down
                          </desc>
                          <path
                            d="M16 6c-.28 0-.53.11-.71.29L10 11.59l-5.29-5.3a1.003 1.003 0 00-1.42 1.42l6 6c.18.18.43.29.71.29s.53-.11.71-.29l6-6A1.003 1.003 0 0016 6z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="bp3-collapse"
                  >
                    <div
                      aria-hidden="true"
                      class="bp3-collapse-body"
                      style="transform: translateY(-undefinedpx);"
                    >
                      <div
                        class="StyledProps--font StyledProps--main StyledProps--background-grey100"
                      >
                        <div
                          class="StyledProps--font StyledProps--main ruleViewerContainer StyledProps--border StyledProps--border-top StyledProps--padding-top-small StyledProps--padding-bottom-small StyledProps--margin-left-large StyledProps--margin-right-large"
                        >
                          <div
                            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main ruleViewContainer StyledProps--padding-top-small StyledProps--padding-bottom-small"
                          >
                            <p
                              class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
                            >
                              Common &gt; Product
                            </p>
                            <p
                              class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi"
                            >
                              IN
                            </p>
                            <p
                              class="StyledProps--font StyledProps--main pillRadius StyledProps--border StyledProps--background-grey200 StyledProps--padding-left-small StyledProps--padding-right-small StyledProps--padding-top-xsmall StyledProps--padding-bottom-xsmall StyledProps--font-variation-tiny"
                            >
                              d
                            </p>
                            <p
                              class="StyledProps--font StyledProps--main pillRadius StyledProps--border StyledProps--background-grey200 StyledProps--padding-left-small StyledProps--padding-right-small StyledProps--padding-top-xsmall StyledProps--padding-bottom-xsmall StyledProps--font-variation-tiny"
                            >
                              e
                            </p>
                            <p
                              class="StyledProps--font StyledProps--main pillRadius StyledProps--border StyledProps--background-grey200 StyledProps--padding-left-small StyledProps--padding-right-small StyledProps--padding-top-xsmall StyledProps--padding-bottom-xsmall StyledProps--font-variation-tiny"
                            >
                              +1
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
